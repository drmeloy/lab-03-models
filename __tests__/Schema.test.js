const Schema = require('../lib/Schema');

describe('Schema tests', () => {
  it('validates an object matching a schema', () => {
    const schema = new Schema({
      name: {
        type: String,
        required: true
      },
      age: {
        type: Number
      },
      weight: {
        type: String
      }
    });

    const cat = {
      name: 'Meowzors',
      age: 6,
      weight: '6 lbs'
    };

    expect(schema.validate(cat)).toEqual({
      name: 'Meowzors',
      age: 6,
      weight: '6 lbs'
    });

  });

  it('throws error on object not matching schema', () => {
    const schema = new Schema({
      name: {
        type: String,
        required: true
      },
      age: {
        type: Number
      },
      weight: {
        type: String
      }
    });

    const cat = {
      age: 'SUP?!',
      weight: 4
    };

    expect(() => schema.validate(cat)).toThrowErrorMatchingSnapshot();
  });

});
