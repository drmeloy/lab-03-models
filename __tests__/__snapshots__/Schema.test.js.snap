// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema tests throws error on object not matching schema 1`] = `"invalid schema >> Error: name is required,Error: Cannot cast >>SUP?!<< to Number"`;
